### PR TITLE
Use login once

### PR DESCRIPTION
**Pull request by innoflash**
This PR aims to fix this issue https://github.com/grosv/laravel-passwordless-login/issues/28

Usage:
```php
// on config.php
'login_use_once'  => true,
```
Or on a AuthenticatableUser that uses the trait:
```php
public function getLoginUseOnceAttribute()
{
    return true;
}
```

I used caching the route for the minutes given.

### CODE REVIEW

**Comment by edgrosvenor** [comment]
@innoflash This is cool. Is it ready to go? It looks good to me but want to make sure everything you think it needs is here. If so let me know and I'll merge it in.

**Comment by innoflash** [comment]
@edgrosvenor hi, last time i tried it was working, i just failed to figure out how your tests are wired but i would have written a test for it.

As far i have tested manually it is working, you can merge and lets see if someone is gonna have an issue with the latest change.

Screenshot from 2020-04-23 17-36-58

I got this error when i tried to wire the tests but its all good.

**Comment by edgrosvenor** [comment]
@innoflash My tests are set up to use orchestra testbench so from the project root you just run `vendor/bin/phpunit`. Your method of testing probably doesn't load the Laravel app setup including the migrations.

**Comment by innoflash** [comment]
Great, i think as of now i will let you handle the testing.

Let me know if i should tweak something.

File: config/config.php [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
-return[
-    'user_model'                => env('LPL_USER_MODEL', 'App\User'),
-    'user_guard'                => env('LPL_USER_GUARD', 'web'),
-    'remember_login'            => env('LPL_REMEMBER_LOGIN', false),
-    'login_route'               => env('LPL_LOGIN_ROUTE', '/magic-login'),
-    'login_route_name'          => env('LPL_LOGIN_ROUTE_NAME', 'magic-login'),
-    'login_route_expires'       => env('LPL_LOGIN_ROUTE_EXPIRES', '30'),
-    'redirect_on_success'       => env('LPL_REDIRECT_ON_LOGIN', '/'),
-    'invalid_signature_message' => env('LPL_INVALID_SIGNATURE_MESSAGE', 'Invalid or Expired link'),
```
Comment (kamiben) [inline config/config.php:11]: Be careful this commits deletes the new config for signature message

File: config/config.php [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+return [
+    'user_model'          => env('LPL_USER_MODEL', 'App\User'),
+    'user_guard'          => env('LPL_USER_GUARD', 'web'),
+    'remember_login'      => env('LPL_REMEMBER_LOGIN', false),
+    'login_route'         => env('LPL_LOGIN_ROUTE', '/magic-login'),
+    'login_route_name'    => env('LPL_LOGIN_ROUTE_NAME', 'magic-login'),
+    'login_route_expires' => env('LPL_LOGIN_ROUTE_EXPIRES', '30'),
+    'redirect_on_success' => env('LPL_REDIRECT_ON_LOGIN', '/'),
+    'login_use_once'      => false,
+
```
Comment (kamiben) [inline config/config.php:13]:
```suggestion
'invalid_signature_message' => env('LPL_INVALID_SIGNATURE_MESSAGE', 'Invalid or Expired link'),
```

File: config/config.php (hunk 1 above)
Comment (innoflash) [reply config/config.php:11]: I think it was fixed by @edgrosvenor when he merged master into this branch.

**Review by kamiben — COMMENTED** [review]
Seems all good for me, I'll test the feature when it's released.

File: config/config.php (hunk 1 above)
Comment (kamiben) [reply config/config.php:11]: ok, thanks ! :)

**Pull request merged**